### ISSUE TITLE
Gradient scaling for hidden state

### ISSUE THREAD

**Issue by mojtabamozaffar** [werner-duvaud/muzero-general#22]
This is not necessarily a bug, but as far as I can see in your code you neglected the hidden state gradient scaling in training the dynamic function as it is suggested by the Muzero paper. Here is a snippet of the paper psudocode: 
`for action in actions:`
`value, reward, policy_logits, hidden_state = network.recurrent_inference(hidden_state, action)`
`predictions.append((1.0 / len(actions), value, reward, policy_logits))`
`hidden_state = scale_gradient(hidden_state, 0.5)`

Is there any specific reason why you ignored it? Actually, when I read the paper I why this scaling "ensures that the total gradient applied to the dynamics function stays constant" as the authors stated. I would be thankful if you share your insight about the effectiveness of this operation.

**Comment by werner-duvaud** [comment]
Hi,

We do a lot of testing about gradient scaling (we also played with gradient clipping). But for the moment we have not seen any big improvement in results or stability.
We don't quite understand why DeepMind do this operation but they surely have a good reason.

 I do not know if someone has explored the question?

We will commit a better gradient scaling once we have finished the tests.
